cluster/etcd_elector.go

```diff
@@ -102,6 +102,16 @@ func (e *etcdLeaderElector) Leader(ctx context.Context) (string, error) {
 	return string(resp.Kvs[0].Value), nil
 }
 
+// IsLeader returns true if the caller is the leader.
+func (e *etcdLeaderElector) IsLeader(ctx context.Context) (bool, error) {
+	resp, err := e.election.Leader(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	return string(resp.Kvs[0].Value) == e.id, nil
```

Comment:
should this be length checked?

Reply:
Added to `Leader()` as well.

I don't think it is needed for either case as we can assume that there is a leader session so the key exists in the DB, but just in case it's good to have this extra check to avoid crashing in case of some unwanted failure.